src/Dropzone/index.jsx

```diff
@@ -0,0 +1,282 @@
+import { useDropzone, ErrorCode } from 'react-dropzone';
+import { fromEvent } from 'file-selector';
+
+import DragError from './DragError';
+import GenericError from './GenericError';
+import UploadProgress from './UploadProgress';
+import DefaultContent from './DefaultContent';
+
+const Dropzone = ({
+  className, accept, minSize, maxSize, validator, progressVariant,
+  uploadCallbacks, errorMessages, axiosConfig, inputComponent,
+}) => {
+  const [isMultipleDragged, setIsMultipleDragged] = useState(false);
+  const [errors, setErrors] = useState([]);
+  const [progress, setProgress] = useState(0);
+  const [fileName, setFileName] = useState(undefined);
+  const [controller, setController] = useState(undefined);
+
+  const {
+    onUploadProgress = () => {},
+    onUploadSuccess = () => {},
+    onUploadCancel = () => {},
+    onUploadError = () => {},
+  } = uploadCallbacks;
+  const {
+    uploadError: uploadErrorMsg,
+    invalidSize: invalidSizeMsg,
+    invalidType: invalidTypeMsg,
+    multipleDragged: multipleDraggedMsg,
+  } = errorMessages;
+
+  const onDragEnter = useCallback(async (e) => {
+    if (errors) {
+      setErrors([]);
+    }
+    const files = await fromEvent(e);
+    if (files && files.length > 1) {
+      setIsMultipleDragged(true);
+    }
+  }, [errors]);
+
+  const onDragLeave = useCallback(() => {
+    if (isMultipleDragged) {
+      setIsMultipleDragged(false);
+    }
+  }, [isMultipleDragged]);
+
+  const onDropRejected = (files) => {
+    if (!isMultipleDragged) {
+      setErrors(files[0].errors.map(error => {
+        switch (error.code) {
+          case ErrorCode.FileTooLarge:
+            return invalidSizeMsg || error.message;
+          case ErrorCode.FileInvalidType:
+            return invalidTypeMsg || error.message;
+          default:
+            return error.message;
+        }
+      }));
+    } else {
+      setIsMultipleDragged(false);
+    }
+  };
+
+  const handleProgressUpload = (progressEvent) => {
+    const percentValue = Math.round((progressEvent.loaded * 100) / progressEvent.total);
+    setProgress(percentValue);
+    onUploadProgress(percentValue, progressEvent);
+  };
+
+  const processUpload = (client, uploadUrl, data) => {
+    const newController = new AbortController();
+    setController(newController);
+
+    const config = {
+      onUploadProgress: handleProgressUpload,
+      signal: newController.signal,
+    };
+
+    client.post(uploadUrl, data, config)
+      .then(response => onUploadSuccess(response))
+      .catch(error => {
+        // check if request has been canceled before treating the exception as an upload error
+        if (error.code !== 'ERR_CANCELED') {
+          setErrors([uploadErrorMsg]);
+          onUploadError(error);
+        }
+      });
+  };
+
+  const onDropAccepted = async (files) => {
+    if (isMultipleDragged) {
+      return;
+    }
+
+    const file = files[0];
+    if (validator) {
+      const customValidationError = await validator(file);
```

Comment:
No, non-async function works too, I've tested it. The reason I added async is basically because of the example with image dimensions validation, there appears to be no way of reading image's dimensions without await / asycs like in the example...

Reply:
Great, thanks for confirming/testing! I think this is fine.